llbcache/resolveimage.go

```diff
@@ -0,0 +1,98 @@
+package llbcache
+
+import (
+	"context"
+	"sync"
+
+	"github.com/moby/buildkit/client/llb"
+	"github.com/opencontainers/go-digest"
+)
+
+// ResolveImageFunc is used to populate the cache for an image.
+type ResolveImageFunc func() (digest.Digest, []byte, error)
+
+// ResolveImageCache provides an interface to lookup resolved images in a cache
+// which allows multiple (and/or parallel) lookups for the same image to be
+// dedup'd and cached.
+type ResolveImageCache interface {
```

Comment:
Any particular reason to abstract behind an interface instead of exporting `resolveImageCache`?

Reply:
Hmm, we could, the interface forces users to use the NewResolveImageCache (which correctly initializes the map).  If we remove the interface, then users could wrongly use `&ResolveImageCache{}`, but that is a "well, just don't do that" type thing.  Either way seems fine to me.

Reply:
I don't mind either way. The general philosophy is to avoid interfaces unless you actually want something to be pluggable, but this seems reasonable.

Reply:
On second thought, I think I will just squash this impl into llblib, leave it private, the code is simplified enough that I don't think it needs its own package now.

Reply:
Updated to squash the llbcache pkg, so interface deleted etc.